tests/test_python_engine.py

```diff
@@ -526,3 +526,55 @@ def evaluate(expr, event):
         self.assertEqual(evaluate("`a.b`", {"a.b": 1}), 1)
         self.assertEqual(evaluate("a.`b.c`[0]", {"a": {"b.c": [1]}}), 1)
         self.assertEqual(evaluate("`!@#$%^&*().`", {"!@#$%^&*().": 1}), 1)
+
+    def test_missing_events(self):
+        """Test the missing event feature."""
+        config = {'flatten': True}
+        events = [Event.from_data(d) for d in [
+            {
+                "event_type": "process",
+                "process_name": "malicious.exe",
+                "unique_pid": "host1-1",
+                "@timestamp": "2023-10-23T00:00:00"
+            },
+            {
+                "event_type": "process",
+                "process_name": "missing.exe",
+                "unique_pid": "host1-1",
+                "@timestamp": "2023-10-23T00:00:00"
+            },
+            {
+                "event_type": "file",
+                "file_name": "suspicious.txt",
+                "unique_pid": "host1-1",
+                "@timestamp": "2023-10-23T00:01:00"
+            }
+        ]]
+
+        # Should return no results since the malicious2.exe event is not missing
+        query = '''
+        sequence by unique_pid with maxspan=1m
+            [ process where process_name == "malicious.exe" ]
+            ![ process where process_name == "missing.exe" ]
+            [ file where file_name == "suspicious.txt" ]
+        '''
+        with elasticsearch_syntax, allow_negation:
+            parsed_query = parse_query(query)
+
+        output = self.get_output(queries=[parsed_query], config=config, events=events)
+
+        self.assertEqual(len(output), 0, "Missing or extra results")
+
+        # Should return results since the second malicious.exe event is missing
+        query = '''
+        sequence by unique_pid with maxspan=1m
```

Comment:
Can you additionally test maxspan for this feature (maybe add a test). I updated the file event to have occurred a day later (`"@timestamp": "2023-11-23T00:01:00"`) and this still found the 3 events

Reply:
Good idea.

Reply:
I was thinking the same thing 😅

Reply:
So the issue was I was using `@timestamp` vs `timestamp`. maxspan works as expected.